services/ml/compilation_delegate_mps.mm

```diff
@@ -50,6 +52,18 @@
     dimensions[3] = extend_channels;
   }
 
+  // The MPSImage will exceed the maximum allowed number of 2048 slices that
+  // need to be reshaped and then reshape to expected dimensions when using. So
+  // the operation of using the input data must be Reshape Operation.
+  if (dimensions[3] > MAX_TEXTURE_ARRAY) {
+    uint32_t times = dimensions[3] / MAX_TEXTURE_ARRAY + 1;
+    // Reshape feature channels with extending width.
+    if (dimensions[3] % times == 0) {
```

Comment:
I suppose we need to find the divisor by searching, e.g.
```
uint32_t times = ceil(dimensions[3]/MAX_TEXTURE_ARRAY);
while (dimensions[3] % times != 0 && times < dimensions[3])
    times++;
```
Need we?

Reply:
Good suggestion, thanks, done.